perl_lib/EPrints/Extras.pm

```diff
@@ -304,7 +304,7 @@ sub render_possible_doi
 
 	$value = "" unless defined $value;
 	if( $value =~ m!^
-			(?:http://(?:dx\.)?doi\.org/)?  # add this again later anyway
+			(?:http(?:s)://(?:dx\.)?doi\.org/)?  # add this again later anyway
```

Comment:
It can be simpler, since the non-capturing group only contains a single character:

`(?:https?://(?:dx\.)?doi\.org/)?`

Incidentally, we already use this in our repository, in a local fork ;)